src/test/typescript-service-helpers.ts

```diff
@@ -1445,6 +1445,114 @@ export function describeTypeScriptService(createService: TypeScriptServiceFactor
 		});
 	});
 
+	describe('getSignatureHelp()', <any> function (this: TestContext) {
+		beforeEach(<any> initializeTypeScriptService(createService, new Map([
+			['file:///a.ts', [
+				'class A {',
+				'	/** foo doc*/',
+				'    foo() {}',
+				'	/** bar doc*/',
+				'    bar(): number { return 1; }',
+				'	/** ',
+				'     * The Baz function',
+				'     * @param num Number parameter',
+				'     * @param text Text parameter',
+				'	  */',
+				'    baz(num: number, text: string): string { return ""; }',
+				'	/** qux doc*/',
+				'    qux: number;',
+				'}',
+				'const a = new A();',
+				'a.baz(32, sd)'
+			].join('\n')],
+			['file:///uses-import.ts', [
+				'import * as i from "./import"',
+				'i.d()'
+			].join('\n')],
+			['file:///import.ts', '/** d doc*/ export function d() {}'],
+			['file:///uses-reference.ts', [
+				'/// <reference path="reference.ts" />',
+				'let z : foo.'
+			].join('\n')],
+			['file:///reference.ts', [
+				'namespace foo {',
+				'	/** bar doc*/',
+				'	export interface bar {}',
+				'}'
+			].join('\n')],
+			['file:///empty.ts', '']
+		])));
+
+		afterEach(<any> shutdownService);
+
+		it('Provides a valid empty response when no signature is found', <any> async function (this: TestContext) {
+			const result = await this.service.getSignatureHelp({
+				textDocument: {
+					uri: 'file:///a.ts'
+				},
+				position: {
+					line: 0,
+					character: 0
+				}
+			});
+			// TODO: activeSignature and activeParameter should allow undefined according to protocol.md
+			assert.deepEqual(result, {
+				signatures: [],
+				activeSignature: 0,
+				activeParameter: 0
+			});
+		});
+
+		it('Provides signature help with parameters in the same file', <any> async function (this: TestContext) {
+			const result = await this.service.getSignatureHelp({
+				textDocument: {
+					uri: 'file:///a.ts'
+				},
+				position: {
+					line: 15,
+					character: 11
+				}
+			});
+			assert.deepEqual(result, {
+				signatures: [
+					{
+						label: 'baz(num, text): string',
+						documentation: 'The Baz function',
+						parameters: [{
+							label: 'num',
+							documentation: 'Number parameter'
+						}, {
+							label: 'text',
```

Comment:
these should include the types of the parameters, like with tsserver:

![image](https://cloud.githubusercontent.com/assets/10532611/24780754/2cf42d6e-1b3a-11e7-9533-399150cc6159.png)